YAKL_atomics.h

```diff
@@ -102,9 +102,91 @@
         ::atomicMax( &update , value );
       }
     #endif
-  #endif
 
-  #ifdef __USE_HIP__
+  #elif defined(__USE_SYCL__)
+    template <typename T, sycl::access::address_space addressSpace =
+	      sycl::access::address_space::global_space>
+    using relaxed_atomic_ref =
+	sycl::ONEAPI::atomic_ref< T,
```

Comment:
SYCL 2020 based on C++20 standards. These are Intel-specific [Fine for now]. 

No flags needed for Intel for C++20.